### PR TITLE
[13_0_X] Addition of 2023 WFs and Fixing NANO step

### PR DESCRIPTION
**Pull request by sunilUIET**
Add 2023 WFs and fix the reco step to store Nano output.
This is a partial backport of https://github.com/cms-sw/cmssw/pull/42069 and https://github.com/cms-sw/cmssw/pull/42047

There is a change in the HLT step as well to have the correct era/conditions for 2022 WFs starting from 139.001

### CODE REVIEW

**Comment by sunilUIET** [comment]
est parameters:

relvals_opt= -w standard
workflows = 141.001,141.002,141.003,141.004,141.005,141.006,141.007,141.008,141.009,141.010,141.011,141.012,141.013,141.031,141.032,141.033,141.034,141.035,141.036,141.037,141.038

**Comment by sunilUIET** [comment]
@cmsbuild please test

**Comment by sunilUIET** [comment]
test parameters:

relvals_opt= -w standard
workflows = 141.001,141.002,141.003,141.004,141.005,141.006,141.007,141.008,141.009,141.010,141.011,141.012,141.013,141.031,141.032,141.033,141.034,141.035,141.036,141.037,141.038

**Comment by sunilUIET** [comment]
@cmsbuild please test

**Comment by sunilUIET** [comment]
+pdmv

**Comment by srimanob** [comment]
A better situation, but I am not sure why we see failure in 139.001 which is RunMinimumBias2021 collision. Any idea, @sunilUIET

**Comment by sunilUIET** [comment]
@srimanob 

I am not sure but one possibility is the HLTDR3_2023 which uses era Run3_2023. It should be era Run3 for 2021 and 2022 WFs.

**Comment by srimanob** [comment]
So, the last fix comes from https://github.com/cms-sw/cmssw/pull/41629, right?

**Comment by sunilUIET** [comment]
> So, the last fix comes from #41629, right? If yes, maybe note in the PR description. Thx.

Apparently yes. I am fixing it

**Comment by sunilUIET** [comment]
@cmsbuild please test

**Comment by sunilUIET** [comment]
@cmsbuild please test

**Comment by sunilUIET** [comment]
@srimanob do see the issue now.

**Comment by perrotta** [comment]
@sunilUIET when you are ready with this backport please don't forget to prepare also the PR to backport in 13_1_X

**Comment by srimanob** [comment]
> @srimanob do see the issue now.

Is this PR ready to go, or you need to fix?

**Comment by sunilUIET** [comment]
@srimanob I fixed it. I was wondering if you still see the issue you mentioned earlier. Sorry I could not check it from the logs

**Comment by srimanob** [comment]
I think if you summarize on 139.001 then we should be good to go. @sunilUIET

**Comment by sunilUIET** [comment]
+pdmv

**Comment by srimanob** [comment]
Sorry @sunilUIET 
Do you get a clear answer on 139.001? If yes, could you please update the PR description, then this PR is good to go. Thanks.

**Comment by sunilUIET** [comment]
> Sorry @sunilUIET Do you get a clear answer on 139.001? If yes, could you please update the PR description, then this PR is good to go. Thanks.

@srimanob it is done

**Comment by perrotta** [comment]
backport of https://github.com/cms-sw/cmssw/pull/42069 
(partial backport; it also includes the bug fixes in https://github.com/cms-sw/cmssw/pull/42047)

**Comment by srimanob** [comment]
+Upgrade

**Comment by perrotta** [comment]
@sunilUIET this still miss the backport in 13_1_X: let wait for it being at least submitted before merging this PR in the older 13_0_X cycle

**Comment by perrotta** [comment]
+1
- partial backport of https://github.com/cms-sw/cmssw/pull/42069
- partial backport of https://github.com/cms-sw/cmssw/pull/41629
- backport of the bug fixes in https://github.com/cms-sw/cmssw/pull/42047)

**Comment by sunilUIET** [comment]
Hi @perrotta I am not fully available this week. I will work on the backport next week.

**Comment by perrotta** [comment]
@sunilUIET all newly added worklows are failing in the CMSSW_13_0_X_2023-07-04-2300 IB
I am going to revert this PR in 13_0_X so that we can build 13_0_10, see #42187

**Comment by srimanob** [comment]
There was a typo `MINIAOOD`, which should be `MINIAOD`.
I clone this PR and fix the typo in https://github.com/cms-sw/cmssw/pull/42191

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2849,9 +2881,13 @@ def gen2021HiMix(fragment,howMuch):
 steps['RECODR3_reHLT_HCALOnlyCPU']=merge([{'-s': 'RAW2DIGI:RawToDigi_hcalOnly,RECO:reconstruction_hcalOnly,DQM:@hcalOnly+@hcal2Only'},steps['RECODR3_reHLT']])
 steps['RECODR3_reHLT_HCALOnlyGPU']=merge([step3_gpu, steps['RECODR3_reHLT_HCALOnlyCPU']])
 
-steps['RECONANORUN3_reHLT']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,PAT,NANO,DQM:@standardDQM+@miniAODDQM+@nanoAODDQM'},steps['RECODR3_reHLT']])
-steps['RECONANORUN3_ZB_reHLT']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,PAT,NANO,DQM:@rerecoZeroBias+@miniAODDQM+@nanoAODDQM'},steps['RECODR3_reHLT']])
-steps['RECOCOSMRUN3_reHLT']=merge([{'--scenario':'cosmics','-s':'RAW2DIGI,L1Reco,RECO,DQM','--datatier':'RECO,DQMIO','--eventcontent':'RECO,DQM'},steps['RECONANORUN3_reHLT']])
+steps['RECONANORUN3_reHLT']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,PAT,NANO,DQM:@standardDQMFakeHLT+@miniAODDQM+@nanoAODDQM','--datatier':'RECO,MINIAOD,NANOAOD,DQMIO','--eventcontent':'RECO,MINIAOOD,NANOEDMAOD,DQM'},steps['RECODR3_reHLT']])
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2884]: Typo here `MINIAOOD`

**Pull request merged**